superset-frontend/src/dataMask/reducer.ts

```diff
@@ -38,24 +38,26 @@ export function getInitialDataMask(id: string): DataMaskWithId {
 
 const dataMaskReducer = produce(
   (draft: DataMaskStateWithId, action: AnyDataMaskAction) => {
-    const oldData = { ...draft };
+    const cleanState = {};
+    const oldState = { ...draft };
     switch (action.type) {
       case UPDATE_DATA_MASK:
         draft[action.filterId] = {
           ...draft[action.filterId],
           ...action.dataMask,
           id: action.filterId,
         };
-        break;
+        return draft;
 
       case SET_DATA_MASK_FOR_FILTER_CONFIG_COMPLETE:
         (action.filterConfig ?? []).forEach(filter => {
-          draft[filter.id] =
-            oldData[filter.id] ?? getInitialDataMask(filter.id);
+          cleanState[filter.id] =
+            oldState[filter.id] ?? getInitialDataMask(filter.id);
```

Comment:
Can't we just do `... = draft[filter.id]...` here instead and remove `oldState`?